src/jvmMain/kotlin/com/sdercolin/vlabeler/video/VideoState.kt

```diff
@@ -0,0 +1,73 @@
+package com.sdercolin.vlabeler.video
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.awt.ComposeWindow
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.sdercolin.vlabeler.audio.PlayerState
+import com.sdercolin.vlabeler.util.toMillisecond
+import java.awt.Container
+
+class VideoState(val debug: Boolean = true) {
+    var width: Dp by mutableStateOf(RememberWidth)
+    var height: Dp = RememberHeight
+
+    var window: ComposeWindow? = null
+    lateinit var container: Container
+
+    var playerState: PlayerState? = null
+    var videoPath: String = ""
+    var currentSampleRate = 0f
+    var embeddedMode: Boolean by mutableStateOf(true)
+
+    val miniVideo = MiniVideo()
+    var afterPopup: Boolean = false
+    val currentTime: Long
+        get() = miniVideo.mediaPlayer.status().time()
+    lateinit var syncOp: (time: Long?) -> Unit
+
+    fun log(message: String): VideoState {
+        if (debug)
+            println("##video## $message")
+        return this
+    }
+
+    fun setPlayerReference(playerState: PlayerState): VideoState {
+        this.playerState = playerState
+        return this
+    }
+
+    fun newestTime(): Long? {
+        return playerState?.framePosition?.let {
+            toMillisecond(it, currentSampleRate).toLong()
+        }
+    }
+
+    fun rememberSize(): VideoState {
```

Comment:
Since `VideoState` is created only once in `AppState`, and `AppState` is basically singleton, I think we don't need `RememberWidth` and `RememberHeight` at all?

Reply:
Done.